app/2.1.x/upgrading.md

```diff
@@ -2,21 +2,202 @@
+   is to prevent the new cluster from generating data that is not understood
+   by the old cluster.
+3. Gradually divert traffic away from your old nodes, and into
+   your 2.1.0 cluster. Monitor your traffic to make sure everything
+   is going smoothly.
+4. When your traffic is fully migrated to the 2.1.0 cluster,
+   decommission your old nodes.
+5. From your 2.1.0 cluster, run: `kong migrations finish`.
+   From this point on, it will not be possible to start
+   nodes in the old cluster pointing to the same datastore anymore. Only run
+   this command when you are confident that your migration
+   was successful. From now on, you can safely make Admin API
+   requests to your 2.1.0 nodes.
+
+**Cassandra**
+
+Due to internal changes, the table schemas used by Kong 2.1.0 on Cassandra
+are incompatible with those used by Kong 2.0.0. Migrating using the usual commands
+`kong migrations up` and `kong migrations finish` will require a small
+window of downtime, since the old and new versions cannot use the
+database at the same time. Alternatively, to keep your previous version fully
+operational while the new one initializes, you will need to transfer the
+data to a new keyspace via a database dump, as described below:
+
+1. Download 2.1.0, and configure it to point to a new keyspace.
+   Run `kong migrations bootstrap`.
+2. Once that finishes running, both the old (pre-2.1) and new (2.1.0)
+   clusters can now run simultaneously, but the new cluster does not
+   have any data yet.
+3. On the old cluster, run `kong config db_export`. This will create
+   a file `kong.yml` with a database dump.
+4. Transfer the file to the new cluster and run
+   `kong config db_import kong.yml`. This will load the data into the new cluster.
+5. Gradually divert traffic away from your old nodes, and into
+   your 2.1.0 cluster. Monitor your traffic to make sure everything
+   is going smoothly.
+6. When your traffic is fully migrated to the 2.1.0 cluster,
+   decommission your old nodes.
+
+##### Installing 2.1.0 on a Fresh Datastore
+
+The following commands should be used to prepare a new 2.1.0 cluster from a
+fresh datastore. By default the `kong` CLI tool will load the configuration
+from `/etc/kong/kong.conf`, but you can optionally use the flag `-c` to
+indicate the path to your configuration file:
+
+```
+$ kong migrations bootstrap [-c /path/to/your/kong.conf]
+$ kong start [-c /path/to/your/kong.conf]
+```
 
-This guide will inform you about breaking changes you should be aware of
-when upgrading, as well as take you through the correct sequence of steps
-in order to obtain a **no-downtime** migration in different upgrade
-scenarios.
 
 ## Upgrade to `2.0.0`
 
 Kong adheres to [semantic versioning](https://semver.org/), which makes a
-distinction between "major", "minor" and "patch" versions. The upgrade path
+distinction between "major", "minor", and "patch" versions. The upgrade path
```

Comment:
```suggestion
distinction between major, minor, and patch versions. The upgrade path
```

don't need quotes. global comment for similar sections. We can set up includes to share content in the future for better consistency and less maintenance.